src/modules/ekf2/ekf2_main.cpp

```diff
@@ -2447,6 +2439,46 @@ float Ekf2::filter_altitude_ellipsoid(float amsl_hgt)
 	return amsl_hgt + _wgs84_hgt_offset;
 }
 
+matrix::SquareMatrix<float, 3> Ekf2::propagate_covariances_from_quat_to_euler(
+	const matrix::Quatf &q,
+	const matrix::SquareMatrix<float, 4> &quat_covariances) const
+{
+	// Jacobian matrix (3x4) containing the partial derivatives of the
+	// Euler angle equations with respect to the quaternions
+	matrix::Matrix<float, 3, 4> G;
+
+	float q1 = q(0);
+	float q2 = q(1);
+	float q3 = q(2);
+	float q4 = q(3);
+
+	// Protect against division by 0
+	float d1 = (2 * q1 * q2 + 2 * q2 * q4) * (2 * q1 * q2 + 2 * q2 * q4) + (-2 * q2 * q2 - 2 * q3 * q3 + 1) *
```

Comment:
You should handle d1, d2, x all the same way. Either math::max or fmaxf in place.

Reply:
done